presto-main/src/test/java/io/prestosql/execution/buffer/BenchmarkBlockSerde.java

```diff
@@ -245,44 +237,38 @@ public Slice getDataSource()
         }
     }
 
-    @State(Thread)
-    public static class LongDecimalNoNullBenchmarkData
+    public abstract static class TypeBenchmarkData<T>
```

Comment:
Is it possible to use composition instead? So you would have single `TypeBenchmarkData` class to which you would inject customizations.

Reply:
JMH has its requirements. I think this is as close to composition as it can get.
I am going to think about it when merging those cases with `BenchmarkColumnReaders` in the following PR